src/utils/error.ts

```diff
@@ -0,0 +1,428 @@
+  constructor (msg: string) {
+    super(msg)
+    this.name = 'MissingContractAddressError'
+  }
+}
+
+export class NoSuchContractError extends AeError {
+  constructor (notExistingContractAddress: string) {
+    super(`Contract with address ${notExistingContractAddress} not found on-chain`)
+    this.name = 'NoSuchContractError'
+  }
+}
+
+export class InactiveContractError extends AeError {
+  constructor (contractAddress: string) {
+    super(`Contract with address ${contractAddress} not active`)
+    this.name = 'InactiveContractError'
+  }
+}
+
+export class BytecodeMismatchError extends AeError {
+  constructor (source: string) {
+    super(`Contract ${source} do not correspond to the bytecode deployed on the chain`)
+    this.name = 'BytecodeMismatchError'
+  }
+}
+
+export class UnknownCallReturnTypeError extends AeError {
+  constructor (returnType: any) {
+    super(`Unknown returnType: ${String(returnType)}`)
+    this.name = 'UnknownCallReturnTypeError'
+  }
+}
+
+export class NodeInvocationError extends AeError {
+  constructor (msg?: string) {
+    msg = msg ?? ''
+    super(`Invocation failed${msg === '' ? '' : `: "${msg}"`}`)
+    this.name = 'NodeInvocationError'
+  }
+}
+
+export class DuplicateContractError extends AeError {
+  constructor () {
+    super('Contract already deployed')
+    this.name = 'DuplicateContractError'
+  }
+}
+
+export class MissingFunctionNameError extends AeError {
+  constructor () {
+    super('Function name is required')
+    this.name = 'MissingFunctionNameError'
+  }
+}
+
+export class InvalidMethodInvocationError extends AeError {
+  constructor (msg: string) {
+    super(msg)
+    this.name = 'InvalidMethodInvocationError'
+  }
+}
+
+export class NotPayableFunctionError extends AeError {
+  constructor (amount: string, fn: string) {
+    super(
+      `You try to pay "${amount}" to function "${fn}" which is not payable. ` +
+      'Only payable function can accept tokens')
+    this.name = 'NotPayableFunctionError'
+  }
+}
+
+export class InvalidSchemaError extends AeError {
+  constructor (msg: string) {
+    super(msg)
+    this.name = 'InvalidSchemaError'
+  }
+}
+
+/* Node communication errors  */
+export class UnsupportedNodeError extends AeError {
+  constructor (nodeVersion: string, NODE_GE_VERSION: string, NODE_LT_VERSION: string) {
+    super(
+      `Unsupported node version ${nodeVersion}. ` +
+      `Supported: >= ${NODE_GE_VERSION} < ${NODE_LT_VERSION}`)
+    this.name = 'UnsupportedNodeError'
+  }
+}
+
+/* Channel originated errors */
+export class ChannelInitializationError extends AeError {
+  constructor (msg: string) {
+    super(msg)
+    this.name = 'ChannelInitializationError'
+  }
+}
+
+export class UnknownChannelStateError extends AeError {
+  constructor () {
+    super('State Channels FSM entered unknown state')
+    this.name = 'UnknownChannelStateError'
+  }
+}
+
+export class ChannelPingTimedOutError extends AeError {

```

Comment:
We can make hierarchy stronger, for example, inheriting ChannelPingTimedOutError from ChannelError and inheriting ChannelError from AeError, but the hierarchy should be reasonable

Reply:
Made following as upper hierarchy errors
```
AccountError
AensError
AeppError
ChannelError
CompilerError
ContractError
CryptographyError
NodeError
SwaggerError
TransactionError
WalletError
```